controllers/autoscalingrunnerset_controller.go

```diff
@@ -34,29 +56,282 @@ type AutoscalingRunnerSetReconciler struct {
+		childNamespace := &corev1.Namespace{}
+		childNamespace.Name = childNamespaceType.Name
+		childNamespace.Namespace = childNamespaceType.Namespace
+		if err := r.Create(ctx, childNamespace); err != nil {
+			kvlog.Error(err, "could not create namespace for runners")
+			return ctrl.Result{}, nil
+		}
+	}
+	// End of reconciliation for RunnerJob namespace.
+
+	roleName := fmt.Sprintf("%v-runner-creator", runnerSet.Spec.RunnerScaleSet)
+
+	// Start of reconciliation for RunnerJob RBAC.
+	if err := r.Get(ctx, childNamespaceType, &rbacv1.Role{}); err != nil {
+		role := &rbacv1.Role{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Role",
+				APIVersion: "rbac.authorization.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      roleName,
+				Namespace: namespaceForManagement,
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					Verbs:           []string{"*"},
+					APIGroups:       []string{"batch"},
+					Resources:       []string{"*"},
+					ResourceNames:   []string{},
+					NonResourceURLs: []string{},
+				},
+			},
+		}
+		if err := r.Create(ctx, role); err != nil {
+			kvlog.Error(err, "could not create role for runner creation")
+			return ctrl.Result{}, nil
+		}
+	}
+
+	if err := r.Get(ctx, childNamespaceType, &rbacv1.RoleBinding{}); err != nil {
+		rolebinding := &rbacv1.RoleBinding{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "RoleBinding",
+				APIVersion: "rbac.authorization.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%v-rolebinding", roleName),
+				Namespace: namespaceForManagement,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					APIGroup:  "",
+					Name:      "default",
+					Namespace: namespaceForManagement,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io/v1",
+				Name:     roleName,
+				Kind:     "Role",
+			},
+		}
+		if err := r.Create(ctx, rolebinding); err != nil {
+			kvlog.Error(err, "could not create rolebinding for runner creation")
+			return ctrl.Result{}, nil
+		}
+	}
+	// End of reconciliation for RunnerJob namespace and RBAC.
 
 	return ctrl.Result{}, nil
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *AutoscalingRunnerSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Pod{}, autoscalingRunnerSetOwnerKey, func(rawObj client.Object) []string {
+
+		groupVersion := actionsv1alpha1.GroupVersion.String()
+
+		// grab the job object, extract the owner...
+		pod, ok := rawObj.(*corev1.Pod)
+		if ok {
+			owner := metav1.GetControllerOf(pod)
+			if owner == nil {
+				return nil
+			}
+
+			// ...make sure it's a Pod...
+			if owner.APIVersion != groupVersion || owner.Kind != "AutoscalingRunnerSet" {
+				return nil
+			}
+
+			// ...and if so, return it
+			return []string{owner.Name}
+		}
+
+		namespace, ok := rawObj.(*corev1.Namespace)
```

Comment:
Not completely sure, but I think that we should create another indexer here for the namespace. We are creating a new informer with the method `IndexField` on object of type `corev1.Pod` so the informer will always fetch the object of type `corev1.Pod` and never the `corev1.Namespace`

We should create another Indexer outside this one for the `corev1.Namespace` in this case

Reply:
Also, once we create another indexer, we can get rid of `, ok` because we know that every time the inline function is called, it is going to have a type we specified as the second argument of `IndexField` function

Reply:
Thanks for catching. That makes sense. Went ahead and made two indexers